### PR TITLE
QA-12458 improve get ref node

### PR DESCRIPTION
**Pull request by Taybou**
## JIRA

https://jira.jahia.org/browse/QA-12458

## Description
Improve get ref node

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrProperty.java [hunk 1]
```diff
@@ -341,7 +341,7 @@ public Double getFloatValue() {
     @GraphQLDescription("GraphQL representation of the node this property references in case the property is single-valued, null otherwise")
     public GqlJcrNode getRefNode() throws GqlJcrUnresolvedNodeReferenceException {
         try {
-            if (property.isMultiple()) {
+            if (property.isMultiple() || property.getType() != PropertyType.REFERENCE || property.getType() != PropertyType.WEAKREFERENCE) {
```
Comment (tdraier) [inline graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrProperty.java:344]: In which case could `(property.getType() != PropertyType.REFERENCE || property.getType() != PropertyType.WEAKREFERENCE)` ever be false ?

Comment (Taybou) [reply graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrProperty.java:344]: no idea, it's what was suggested in the ticket.

Comment (Taybou) [reply graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrProperty.java:344]: And when I tested in the node properties:
- the j:nodename for example it's of type reference

Comment (tdraier) [reply graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrProperty.java:344]: Hint : it's not possible to have property.getType() (a number) equals to PropertyType.REFERENCE and PropertyType.WEAKREFERENCE at the same time. So the opposite condition will always be true, hence the condition is broken (can be simplified to if (true) ) and you will never get a value anymore when calling getRefNode

**Pull request merged**